src/h3lib/lib/h3Index.c

```diff
@@ -1052,3 +1052,113 @@ H3Error H3_EXPORT(getPentagons)(int res, H3Index *out) {
  *         a Class II grid.
  */
 int isResolutionClassIII(int res) { return res % 2; }
+
+/**
+ * Returns the position of the cell within an ordered list of all children of
+ * the cell's parent at the specified resolution
+ */
+H3Error H3_EXPORT(cellToChildPos)(H3Index child, int parentRes, int64_t *out) {
+    int childRes = H3_GET_RESOLUTION(child);
+    // Get the parent at res. This will catch any resolution errors
+    H3Index parent;
+    H3Error parentError = H3_EXPORT(cellToParent(child, parentRes, &parent));
+    if (parentError) {
+        return parentError;
+    }
+    int parentIsPentagon = H3_EXPORT(isPentagon)(parent);
+
+    // Walk up the resolution digits, incrementing the index
+    *out = 0;
+    if (parentIsPentagon) {
+        // Pentagon logic. Pentagon parents skip the 1 digit, so the offsets are
+        // different from hexagons
+        for (int res = childRes; res > parentRes; res--) {
+            // It shouldn't be possible to get an error here, so we don't check
+            H3_EXPORT(cellToParent(child, res - 1, &parent));
+            parentIsPentagon = H3_EXPORT(isPentagon)(parent);
+            int rawDigit = H3_GET_INDEX_DIGIT(child, res);
+            int digit =
+                parentIsPentagon && rawDigit > 0 ? rawDigit - 1 : rawDigit;
+            if (digit != CENTER_DIGIT) {
+                int64_t hexChildCount = _ipow(7, childRes - res);
+                // The offset for the 0-digit slot depends on whether the
+                // current index is the child of a pentagon. If so, the offset
+                // is based on the count of pentagon children, otherwise,
+                // hexagon children.
+                *out += (parentIsPentagon
+                             ?  // pentagon children. See the explanation
+                                // for getNumCells in h3api.h.in
+                             1 + (5 * (hexChildCount - 1)) / 6
+                             :  // one hexagon's children
+                             hexChildCount) +
+                        // the other hexagon children
+                        (digit - 1) * hexChildCount;
+            }
+        }
+    } else {
+        // Hexagon logic. Offsets are simple powers of 7
+        for (int res = childRes; res > parentRes; res--) {
+            *out += H3_GET_INDEX_DIGIT(child, res) * _ipow(7, childRes - res);
+        }
+    }
+
+    return E_SUCCESS;
+}
+
+/**
+ * Returns the child cell at a given position within an ordered list of all
+ * children at the specified resolution */
+H3Error H3_EXPORT(childPosToCell)(int64_t childPos, H3Index parent,
+                                  int childRes, H3Index *child) {
+    if (childRes < 0 || childRes > MAX_H3_RES) {
```

Comment:
This would also be a good place for `testcase` from #720 to ensure boundary conditions work as expected.